shared/src/types/vote_extensions/validator_set_update.rs

```diff
@@ -0,0 +1,297 @@
+//! Contains types necessary for processing validator set updates
+//! in vote extensions.
+
+pub mod encoding;
+
+use std::collections::HashMap;
+
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use encoding::{AbiEncode, Encode, Token};
+use ethabi::ethereum_types as ethereum;
+use num_rational::Ratio;
+
+use crate::ledger::pos::types::{Epoch, VotingPower};
+use crate::proto::Signed;
+use crate::types::address::Address;
+use crate::types::key::common::{self, Signature};
+use crate::types::key::{SigScheme, VerifySigError};
+
+// the namespace strings plugged into validator set hashes
+const BRIDGE_CONTRACT_NAMESPACE: &str = "bridge";
+const GOVERNANCE_CONTRACT_NAMESPACE: &str = "governance";
+
+/// Wrapper type for [`ethereum::Address`]
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct EthAddr(pub ethereum::Address);
```

Comment:
We already have the type in `src/types/ethereum_events` except that we call the `to_fixed_bytes` method on the `ethereum::Address` type to get the raw bytes.